tkg/client/utils.go

```diff
@@ -445,3 +446,19 @@ func (c *TkgClient) ensureClusterTopologyConfiguration() {
 	log.V(6).Infof("Setting %v to %q", constants.ConfigVariableClusterTopology, clusterTopologyValueToSet)
 	c.TKGConfigReaderWriter().Set(constants.ConfigVariableClusterTopology, clusterTopologyValueToSet)
 }
+
+// checkTkrAvailability get tkr version from tkrVersionCli and check if tkr is available in tkrResourceCli
+func checkTkrAvailability(tkrVersionCli, tkrResourceCli clusterclient.Client) error {
+	tkr, err := tkrVersionCli.GetClusterResolvedTanzuKubernetesRelease()
```

Comment:
This function might not work since its trying to get the TKr from Cluster object. But in your case the Cluster is not yet created. So I am not sure if it can get the TKr.

Reply:
ahh, you are right. 
do you have any recommended way to check tkr resource creation?
found a func `GetResourceList` but it requires clustername as well.